src/components/ViewContainer/ViewContainer.js

```diff
@@ -0,0 +1,174 @@
+import React, {
+  Fragment,
+  memo,
+  useState,
+  useRef,
+} from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import { noop } from 'lodash';
+
+import {
+  Callout,
+  ConfirmationModal,
+} from '@folio/stripes/components';
+import { buildUrl } from '@folio/stripes/smart-components';
+
+import { getCRUDActions } from './getCRUDActions';
+import {
+  ExceptionModal,
+  networkMessage,
+} from '../index';
+
+import sharedCss from '../../shared.css';
+
+export const ViewContainer = memo(props => {
+  const {
+    entityKey,
+    children,
+    mutator,
+    location,
+    history,
+    match: { path },
+    selectRecord,
+  } = props;
+
+  const fullWidthContainerRef = useRef();
+  const calloutRef = useRef();
+
+  const [showExceptionModal, setShowExceptionModal] = useState(false);
+  const [showRestoreModal, setShowRestoreModal] = useState(false);
+  const [restoreInProgress, setRestoreInProgress] = useState(false);
+
+  const CRUDActions = getCRUDActions({
+    entityKey,
+    mutator,
+    onSuccess: networkMessage('success', entityKey, calloutRef),
+    onError: networkMessage('error', entityKey, calloutRef),
+  });
+
+  const transitionToParams = params => {
+    history.push(buildUrl(location, params));
+  };
+
+  const onShowRestoreModal = () => {
+    setShowRestoreModal(true);
+  };
+
+  const onHideRestoreModal = () => {
+    setRestoreInProgress(false);
+    setShowRestoreModal(false);
+  };
+
+  const onUpdateSuccess = (record, dispatch, properties) => {
+    const { reset: resetForm } = properties;
+
+    resetForm();
+    transitionToParams({
+      _path: `${path}/view/${record.id}`,
+      layer: null,
+    });
+  };
+
+  const deselectOnDelete = (recordId, selectItem, selectedRecords) => {
+    const isRecordSelected = selectedRecords.has(recordId);
+
+    if (isRecordSelected) {
+      selectItem(recordId);
+    }
+  };
+
+  const onDeleteSuccess = record => {
+    transitionToParams({
+      _path: `${path}/view`,
+      layer: null,
+    });
+    selectRecord(record.id);
+  };
+
+  const onDeleteError = (record, error) => {
+    if (error.status === 409) {
+      setShowExceptionModal(true);
+    }
+  };
+
+  const onRestoreSuccess = record => {
+    transitionToParams({
+      _path: `${path}/view`,
+      layer: null,
+    });
+    onHideRestoreModal();
+  };
+
+  const onRestoreError = (record, error) => {
+    onHideRestoreModal();
+  };
+
+  const restoreDefaults = async () => {
+    if (restoreInProgress) {
+      return;
+    }
+
+    const record = {};
+
+    try {
+      setRestoreInProgress(true);
+
+      await restoreDefaults.POST({});
+
+      onRestoreSuccess();
+    } catch (error) {
+      onRestoreError(record, error);
+    }
+  };
+
+  return (
+    <Fragment>
+      {children({
+        fullWidthContainer: fullWidthContainerRef.current,
+        calloutRef: calloutRef.current,
+        handleCreateSuccess: onUpdateSuccess,
```

Comment:
Could you please rename functions to start with "handle" and not "on". Usually, event property name starts with "on" and the callback name starts with "handle" (e. g. `<button onClick={handleClick}>`)

Reply:
Done